worker/metrics/metricspusherworker.go

```diff
@@ -0,0 +1,12 @@
+package metrics
+
+import (
+	"github.com/trustwallet/go-libs/metrics"
+	"github.com/trustwallet/go-libs/worker"
+)
+
+func NewMetricsPusherWorker(options *worker.WorkerOptions, pusher metrics.Pusher) worker.Worker {
+	w := worker.InitWorker("metrics_pusher", options, pusher.Push)
+	w.(worker.StoppableWorker).WithStop(pusher.Close)
```

Comment:
InitWorker returns a 'Worker interface' and it's not guaranteed that it implements the 'StoppableWorker interface'. I suggest 'InitWorker' to return a struct instead of an interface to solve this.